src/components/selectionBoxLayer.ts

```diff
@@ -10,10 +10,18 @@ export module Components {
       topLeft: { x: 0, y: 0 },
       bottomRight: { x: 0, y: 0 }
     };
+    private _boxDataBounds: Bounds = {
```

Comment:
What's the difference between these `Bounds` and the other one? If these are in data-space, then a more descriptive name should be used (the "data" here is ambiguous). In addition, `Bounds` are defined as containing `Point`s which contain `number`s, which means you couldn't use it to hold other kinds of data (like `Date`s).